hw/ip/otbn/doc/_index.md

```diff
@@ -604,13 +623,41 @@ When it executes this instruction, OTBN:
   </tbody>
 </table>
 
-## Device Interface Functions (DIFs)
 
-{{< dif_listing "sw/device/lib/dif/dif_otbn.h" >}}
+# Writing OTBN applications {#writing-otbn-applications}
 
-## Register Table
+OTBN applications are (small) pieces of software written in OTBN assembly.
+The full instruction set is described in the [ISA manual]({{< relref "hw/ip/otbn/doc/isa" >}}), and example software is available in the `sw/otbn` directory of the OpenTitan source tree.
 
-{{< registers "hw/ip/otbn/data/otbn.hjson" >}}
+## Toolchain support
+
+OTBN comes with a toolchain consisting of an assembler, a linker, and helper tools such as objdump.
+The toolchain wraps a RV32 GCC toolchain and supports many of its features.
+
+The following tools are available:
+* `otbn-as`: The OTBN assembler.
+* `otbn-ld`: The OTBN linker.
+* `otbn-objdump`: objdump for OTBN.
+
+Other tools from the RV32 toolchain can be used directly, such as objcopy.
+
+## Passing of data between the host CPU and OTBN {#writing-otbn-applications-datapassing}
+
+Passing data between the host CPU and OTBN is done through the data memory (DMEM).
+No standard or required calling convention exists, every application is free to pass data in and out of OTBN in whatever format it finds convenient.
+All data passing must be done as long as OTBN is not running, as indicated by the {{< regref "STATUS.busy" >}} bit; during the OTBN operation both the instruction and the data memory are inaccessible from the host CPU.
```

Comment:
Grammar nit: This should be "... must be done when OTBN is not running".

Here, "as long as" sounds like the host processor has to keep furiously passing data until it finally starts OTBN and can collapse back in its chair with a sigh of relief :-)